pkg/ccl/backupccl/restore_planning.go

```diff
@@ -1170,6 +1185,62 @@ func rewriteSchemaDescs(schemas []*schemadesc.Mutable, descriptorRewrites DescRe
 
 		sc.ID = rewrite.ID
 		sc.ParentID = rewrite.ParentID
+
+		if err := rewriteSchemaChangerState(
+			sc.DeclarativeSchemaChangerState, descriptorRewrites,
+		); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func rewriteSchemaChangerState(
```

Comment:
I'm tempted to say rewriteSchemaChangerState should be in an sc package. 


I mean, I'm even tempted to say all these existing rewrite methods that dig into catalog elements and replace one ID with another should move to an sc package that restore just calls into, but I understand that might be out of scope here, so I guess this adding one here along side keeps it consistent.

Reply:
Sure

Reply:
Well, I thought about it and a lot of symbols would have to move. I directionally agree, but not for this PR.